### PR TITLE
Make: Add changelog script for release-notes

### PR DESCRIPTION
**Pull request by titom73**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [x] Other (please describe): Changelog script

## Proposed changes

Add a script to build release-notes and changelog based on commit and [conventionalcommit](https://www.conventionalcommits.org/en/v1.0.0/) syntax

It requires `ZSH` shell to be executed

- CLI output

```bash
# Script with output in stdout
zsh changelog.sh v3.0.0rc1 v2.2.1

v3.0.0rc1

Features:

 - f79978e [eos_cli_config]     Ensure required directories are present #1044
 - ca121ae [eos_cli_config_gen] Add OSPF timers to router ospf (#1070)
 - 032b6bb [eos_cli_config_gen] Add single-connection to tacacs (#1095)
 - b64bfa5 [eos_cli_config_gen] Implement OSPF inter area filtering (#1076)
 - 1bc5dea [eos_designs]        Implement tag for documentation (#1053)
 - 1e9c100 [eos_designs]        Refactor topology data models for AVD v3.0 (#1000)

Documentation:

 - 50df169 [contribution]       Document conventional commit approach (#1072)
 - 12b4619 [eos_cli_config_gen] Storm control on port channels (#1086)
 - be42308 [eos_designs]        Fix link
 - ffd8734 [mkdocs]             Update incorrect link to topology diagram (#1075)
```

- Markdown output

```
# Script with MD output
$ zsh changelog.sh v3.0.0rc1 v2.2.1 md > CHANGELOG.md
```

## Todo:

- [x] Document Conventional syntax for AVD
- [x] Add GH Action to lint PR

## How to test

See above

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

**Comment by titom73** [comment]
Add PGH actions to enforce PR title with conventional syntax

If PR title is not correct:

- Add tag for easy filtering
- Add comment so user receives notification

image

File: .github/workflows/pr_title.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            upgrade_tools
+            plugins
+            mkdoc
+            contribution
+            how-to
+            actions
+            molecule
+            ansible
+            github
+            requirements
```
Comment (titom73) [inline .github/workflows/pr_title.yml:61]: Add scope other for undefined scope and set scope to True

File: ansible_collections/arista/avd/docs/contribution/overview.md [hunk 2]
```diff
@@ -129,22 +129,41 @@ The issue will be reviewed by a moderator after submission and the appropriate l
...
+    * `{{ role name }}`: AVD role impacting by PR. __Required__ for `Feat`, `Cut` and `Fix` types
+    * `plugins`: To use when AVD plugin is impacted by PR. __Required__ for `Feat`, `Cut` and `Fix` types
+    * `requirements`: To use when using `Bump` type and when any AVD requirement is updated
+    * `mkdoc`: represents generic documentation published on www.avd.sh
+    * `contribution`: documentation related to contribution
+    * `how-to`: documentation in the How to section of avd
+    * `actions`: represents Github Actions update
+    * `molecule`: represents Molecule CI update
+    * `ansible`: For any ansible information update like galaxy.yml or ansible requirements
+    * `github`: For any content related to Github processes
```
Comment (titom73) [inline ansible_collections/arista/avd/docs/contribution/overview.md:163]: Add `other` scope

**Comment by xaviramon** [comment]
I am not really able to follow the entire code, but tested and works well. 
For some reason, first PR number under "Features" and "Documentation" blocks appear always without parenthesis:
```
Features:

 - f79978e [eos_cli_config]     Ensure required directories are present #1044
 ```
```
Documentation:

 - 276cde2                      Add CI badge in README file #1109
```
vs
```
Other changes:

 - 0813eb8 [action]             CI: Enable Concurrency management in workflow (#1034)
```

This is present in all 3 styles of output (raw, text and md). Could we fix that for document integrity?

**Comment by titom73** [comment]
It seems it is fixed:

```bash
v3.0.0rc1-15-gdce79a23

Features:

 - e0d10bb [eos_cli_config_gen] Add RADIUS Source-Interface (#1118)
 - 08c8359 [eos_cli_config_gen] Add support for redistribution of bgp routes into ospf (#1082)
 - f73cfdc [eos_cli_config_gen] Create maintenance configuration (#1020)
 - daf4b11 [eos_cli_config_gen] Http client source interface (#1119)
 - dce79a2 [eos_designs]        Add support for setting VXLAN Source interface (#1149)
 - 1793c20 [plugin]             Add var_type test to defined plugin (#1148)
 - 492a45d [plugin]             Support natural sort based on item key values (#1147)

Bug fixes:

 - f3b4cac [eos_cli_config_gen] AAA authentication on console fix (#1133)
 - bc41cbf [eos_designs]        Add protection on switch_filter_tenants (#1122)
 - fcbec07 [eos_designs]        Make options under SVI `ip_helpers` optional. (#1127)
 - 71e1697 [upgrade_tools]      Avoid generating excess upgrade vars (#1135)

Other changes:

 - 46fc495 [issue_template]     Build system: Instantiate new Issue template (#1128)
```

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**